frame/contracts/src/wasm/mod.rs

```diff
@@ -145,28 +186,38 @@ where
 		let mut runtime = Runtime::new(
 			&mut ext,
 			input_data,
-			&self.schedule,
 			memory,
 			gas_meter,
 		);
 
 		// Instantiate the instance from the instrumented module code and invoke the contract
 		// entrypoint.
-		let result = sp_sandbox::Instance::new(&exec.prefab_module.code, &imports, &mut runtime)
-			.and_then(|mut instance| instance.invoke(exec.entrypoint_name, &[], &mut runtime));
+		let result = sp_sandbox::Instance::new(&self.code, &imports, &mut runtime)
+			.and_then(|mut instance| instance.invoke(function.identifier(), &[], &mut runtime));
 		runtime.to_execution_result(result)
 	}
+
+	fn code_hash(&self) -> &CodeHash<T> {
+		&self.code_hash
+	}
+
+	fn occupied_storage(&self) -> u32 {
+		use sp_std::mem::size_of;
+		let len = self.original_code_len
+			.saturating_add(self.code.len() as u32)
+			.saturating_add(size_of::<PrefabWasmModule<T>>() as u32);
```

Comment:
actually the struct make use of compact encoding of integers and some fields are skipped, thus the size is different of the struct is quite different from its encoding.

maybe you can use `Encode::encoded_size` it is not super optimized but it could be fine.

At some point I agree code len makes other fields negligeable, but then at least we should have a comment.

Reply:
Doesn't `Encode::encoded_size` perform non trivial work? Basically doing a dry-run encoding. Maybe `Encode::size_hint` would be more appropriate?

Reply:
I think `Encode::size_hint` is pretty inaccurate
yes `Encode::encoded_size` is non trivial work.
the most efficient and correct at the same time would be to implement here a function which calculate the size directly. (and that need to be updated when PrefabWasmModule changes, though some tests can catch this.)

Reply:
I opted for the most simple solution: Disregard the struct size as it is dominated by the code size and added a comment about that.